datalad/support/network.py

```diff
@@ -1103,4 +1152,60 @@ def download_url(url, dest=None, overwrite=False):
         return providers.fetch(url)
 
 
+def local_path2url_path(local_path: str,
+                        auto_resolve: bool = False
+                        ) -> str:
+    """Convert a local path into a URL path component"""
+    if not local_path:
+        if not auto_resolve:
+            raise ValueError("cannot convert empty local path to URL path")
+        local_path = os.getcwd()
+
+    url = urlparse(Path(local_path).as_uri())
+    if url.netloc:
+        raise ValueError(
+            f"cannot convert remote path to an URL path: {local_path}")
+    return url.path
+
+
+def url_path2local_path(url_path: str | PurePosixPath,
+                        make_absolute: bool = True,
+                        ) -> str | Path:
+
+    if isinstance(url_path, PurePosixPath):
+        return_path = True
+        url_path = str(url_path)
+    else:
+        return_path = False
+
+    if not url_path or not url_path.startswith("/"):
+        # We expect a 'path-absolute' as defined in RFC 3986, therefore the
+        # path must begin with a slash.
+        raise ValueError(
+            f"url path does not start with '/': {url_path}, and is therefore "
+            f"not an absolute-path as defined in RFC 8089")
+
+    if url_path.startswith("//"):
+        # We expect a 'path-absolute' as defined in RFC 3986, therefore the
+        # first segment must not be empty, i.e. the path must not start with
+        # two or more slashes.
+        raise ValueError(
+            f"url path has empty first segment: {url_path}, and is therefore "
+            f"not an absolute-path as defined in RFC 8089")
+
+    return (
+        Path(url2pathname(url_path))
+        if return_path
+        else url2pathname(url_path)
+    )
+
+
+def quote_path(path: str, safe: str = "/") -> str:
+    """quote the path component of a URL, takes OS specifics into account"""
```

Comment:
What is safe?

Reply:
It is handed down to `urllib.parse.quote()` from the documentation of `quote`:

```
The optional safe parameter specifies additional ASCII characters that should not be quoted — its default value is '/'.
```

Reply:
doesn't describe that to me here, describe that in the docstring above please, even if at least as an argument to be passed into `urllib.parse.quote`